### PR TITLE
Feat: Post edit and delete

### PR DESCRIPTION
**Pull request by anandsr21harshit**
- User can now delete his own post
- User can also edit the post

### CODE REVIEW

**Review by mansipadgelwar — APPROVED** [review]
Mentioned functionality is working as expected.
- I am able to edit and delete a post
- Good use of reducers
Good to merge

File: src/frontend/components/Header/Header.jsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ function Header() {
 
         {/* for desktop view */}
         {token && <ul className={HeaderCSS.desktop_menu}>
-          <li><Link to={"/home"}><img src={home} alt="Home" /></Link></li>
+          <li><Link to={"/"}><img src={home} alt="Home" /></Link></li>
           <li> <img src={bookmark} alt="" /></li>
           <li><img src={explore} alt="" /></li>
           <li><Link to={"/profile"}><img src={user} alt="Profile" /></Link></li>
```
Comment (manish591) [inline src/frontend/components/Header/Header.jsx:36]: Curly braces are not required around `to` here

File: src/frontend/features/post/postSlice.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
+import axios from "axios"
+import {createSlice, createAsyncThunk} from "@reduxjs/toolkit"
+import { getAllPostsService, addPostService, editPostService, deletePostService } from "../../service/postsService";
+
+export const getAllPosts = createAsyncThunk(
+    "posts/getAllPosts",
+    async (_,thunkAPI)=>{
```
Comment (manish591) [inline src/frontend/features/post/postSlice.js:7]: You can destructure `rejectWithValue` here

**Review by manish591 — APPROVED** [review]
The code is looking good. In some places, I found your code formatting is not consistent. So, use prettier.

**Pull request merged**